File: backend/PlugPlay/PlugPlay.Api/Dto/Product/ReviewDto.cs
using PlugPlay.Domain.Entities;

namespace PlugPlay.Api.Dto.Product;

public record ReviewDto
{
    public int Id { get; set; }

    public int ProductId { get; set; }

    public int? UserId { get; set; }

    public int Rating { get; set; }

    public string Comment { get; set; }

    public UserDto UserDto { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public static ReviewDto MapReview(Review review)
    {
        return new ReviewDto{
            Id = review.Id,
            ProductId = review.ProductId,
            UserId = review.UserId,
            Rating = review.Rating,
            Comment = review.Comment,
            UserDto = UserDto.MapUser(review.User),
            CreatedAt = review.CreatedAt,
            UpdatedAt = review.UpdatedAt};
    }
}

File: mobile/app/src/main/java/com/plugplay/plugplaymobile/data/model/ReviewDto.kt
package com.plugplay.plugplaymobile.data.model

import com.google.gson.annotations.SerializedName

data class ReviewDto(
    @SerializedName("id") val id: Int,
    @SerializedName("rating") val rating: Int,
    @SerializedName("comment") val comment: String?,
    @SerializedName("createdAt") val createdAt: String,
    @SerializedName("userDto") val user: ReviewUserDto?
)

data class ReviewUserDto(
    @SerializedName("id") val id: Int,
    @SerializedName("firstName") val firstName: String?,
    @SerializedName("lastName") val lastName: String?
)